wfe-core/src/test/java/ru/runa/wfe/chat/sender/SessionMessageSenderTest.java

```diff
@@ -2,83 +2,116 @@
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.anySet;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.verifyZeroInteractions;
-import static org.mockito.Mockito.when;
 
 @RunWith(MockitoJUnitRunner.class)
 public class SessionMessageSenderTest {
-
-    @Mock
-    private Session session;
-    @Mock
-    private RemoteEndpoint.Basic basic;
-    @Mock
-    private MessageBroadcast dto;
     @Mock
     private MailMessageSender mailMessageSender;
     @Mock
     private ObjectMapper chatObjectMapper;
     @InjectMocks
     private SessionMessageSender sessionMessageSender;
-
-    @Before
-    public void init() {
-        when(session.getBasicRemote()).thenReturn(basic);
-        final Map<String, Object> sessionUserProperties = singletonMap("user", createUser());
-        when(session.getUserProperties()).thenReturn(sessionUserProperties);
-    }
+    private Set<SessionInfo> sessionsSet = Collections.newSetFromMap(new ConcurrentHashMap<>());
 
     @Test
-    public void whenSessionIsNotNull_thenMessageSent() throws IOException {
+    public void whenSessionsIsNotEmpty_thenMessageSent() throws IOException {
+        RemoteEndpoint.Basic basic = mock(RemoteEndpoint.Basic.class);
+        MessageBroadcast dto = mock(MessageBroadcast.class);
+
+        Session session = mock(Session.class);
+        sessionsSet.add(new SessionInfo(session));
+
+        when(session.getBasicRemote()).thenReturn(basic);
         when(chatObjectMapper.writeValueAsString(dto)).thenReturn("testContent");
 
-        sessionMessageSender.handleMessage(dto, of(session));
+        sessionMessageSender.handleMessage(dto, sessionsSet);
 
         verify(basic).sendText(eq("testContent"));
     }
 
     @Test
-    public void whenSessionIsNotNullAndSendError_thenSendDelegated() throws IOException {
+    public void whenSessionSendError_thenSendDelegated() throws IOException {
+        RemoteEndpoint.Basic basic = mock(RemoteEndpoint.Basic.class);
+        MessageBroadcast dto = mock(MessageBroadcast.class);
+
+        Session session = mock(Session.class);
+        sessionsSet.add(new SessionInfo(session));
+
+        when(session.getBasicRemote()).thenReturn(basic);
         doThrow(new IOException()).when(basic).sendText(notNull());
         when(chatObjectMapper.writeValueAsString(dto)).thenReturn("testContent");
 
-        sessionMessageSender.handleMessage(dto, of(session));
+        sessionMessageSender.handleMessage(dto, sessionsSet);
 
         verify(basic).sendText(eq("testContent"));
-        verify(mailMessageSender).handleMessage(notNull(), eq(empty()));
+        verify(mailMessageSender).handleMessage(notNull(), anySet());
+    }
+
+    @Test
+    public void whenOneOfFewSessionsSendError_thenDontSendDelegated() throws IOException {
```

Comment:
Тест скорее покрывает, что "если хотя бы 1 отправка успешна, то делегат не вызывался", чем "если 1 отправка с ошибкой, то делегат не вызывался".
Поэтому

```suggestion
    public void whenAnySessionSendSuccess_thenDelegateNotInvoked() throws IOException {
```